awx/main/scheduler/task_manager.py

```diff
@@ -410,29 +359,33 @@ def gen_dep_for_inventory_update(self, inventory_task):
         created_dependencies = []
         if inventory_task.source == "scm":
             invsrc = inventory_task.inventory_source
-            if not invsrc.source_project.scm_update_on_launch:
+            project = self.all_projects.get(invsrc.source_project_id, None)
+            if not project:
                 return created_dependencies
 
-            latest_src_project_update = self.get_latest_project_update(invsrc.source_project_id)
-            if self.should_update_related_project(inventory_task, latest_src_project_update):
-                latest_src_project_update = self.create_project_update(inventory_task, project_id=invsrc.source_project_id)
+            latest_src_project_update = project.project_updates.filter(job_type='check').order_by("-created").first()
+            if self.should_update_again(latest_src_project_update, project.scm_update_cache_timeout):
+                latest_src_project_update = self.create_project_update(inventory_task, project)
                 created_dependencies.append(latest_src_project_update)
             self.add_dependencies(inventory_task, [latest_src_project_update])
-            latest_src_project_update.scm_inventory_updates.add(inventory_task)
         return created_dependencies
 
     @timeit
     def generate_dependencies(self, undeped_tasks):
         created_dependencies = []
+        processed_ids = []
         for task in undeped_tasks:
+            if task.dependencies_processed:
+                continue
+            processed_ids.append(task.id)
```

Comment:
@fosterseth I added this because we call `generate_dependencies` an extra time for in case of deps-of-deps, passing in deps. In most cases those deps do not need to have their own deps generated. I will trust my optimizations to create_unified_job, and if a dependent project update has `dependencies_processed` as `False` then obviously we would like to exclude it from this loop for performance and to avoid errors from re-processing.